internal/controller/appstudio-pipeline-sa-watcher-controller.go

```diff
@@ -96,6 +98,14 @@ func (r *AppstudioPipelineServiceAccountWatcherReconciler) Reconcile(ctx context
 		log.Error(err, "failed to list ImageRepositories in namespace", l.Action, l.ActionView)
 		return ctrl.Result{}, err
 	}
+	// Wait all Image Repositories are provisioned
+	for _, imageRepository := range imageRepositoryList.Items {
+		if imageRepository.Status.State == "" {
+			// Wait Image Controller to finish
+			log.Info("Waiting for Image Repository provision", "ImageRepositoryName", imageRepository.Name)
+			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+		}
+	}
```

Comment:
For my understanding, why do we need to wait here? Do we depend on the owner references in a way that can't be worked around by using the labels? (Similarly to what we're adding to the `isComponentImageRepositorySecret` function)

Reply:
The status needs to be checked first, because checking labels is sorta workaround.